pkg/cmd/extension/http.go

```diff
@@ -19,26 +19,25 @@ func hasScript(httpClient *http.Client, repo ghrepo.Interface) (hs bool, err err
 	url := ghinstance.RESTPrefix(repo.RepoHost()) + path
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return
+		return false, err
 	}
 
 	resp, err := httpClient.Do(req)
 	if err != nil {
-		return
+		return false, err
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode == 404 {
-		return
+		return false, repositoryNotFoundErr
```

Comment:
I am a bit worried that this is not an equivalent check to the repository not existing. Technically, the repository can exist but if it does have a top level script of the same name as the repository then it will return this 404 and the error message will be incorrect. I am wondering if we should just add an additional API check to `manager#Install` method that just does the repo exist check. It is an additional API call but installing will not necessarily ever be a fast experience since it requires a download.

Reply:
@samcoe moving the check up higher makes sense to make sure the error message is consistent. Let me see if I can make that change here

Reply:
@samcoe ran into a small snag here https://github.com/cli/cli/pull/6425/files#r994185555